imgui-SFML.cpp

```diff
@@ -686,29 +688,85 @@ void Image(const sf::RenderTexture& texture, const sf::Vector2f& size, const sf:
 /////////////// Image Overloads for sf::Sprite
 
 void Image(const sf::Sprite& sprite, const sf::Color& tintColor, const sf::Color& borderColor) {
-    sf::FloatRect bounds = sprite.getGlobalBounds();
-    Image(sprite, sf::Vector2f(bounds.width, bounds.height), tintColor, borderColor);
+	Image(sprite, sprite.getGlobalBounds().getSize(), sf::Transform::Identity, tintColor, borderColor);
 }
 
 void Image(const sf::Sprite& sprite, const sf::Vector2f& size, const sf::Color& tintColor,
-           const sf::Color& borderColor) {
-    const sf::Texture* texturePtr = sprite.getTexture();
-    // sprite without texture cannot be drawn
-    if (!texturePtr) {
+    const sf::Color& borderColor) {
+	Image(sprite, size, sf::Transform::Identity, tintColor, borderColor);
+}
+
+void Image(const sf::Sprite& sprite, const sf::Vector2f& size, const sf::Transform& additionalTransform,
+    const sf::Color& tintColor, const sf::Color& borderColor)
+{
+    const sf::IntRect& textureRect = sprite.getTextureRect();
+
+    // sprite without texture on empty frame cannot be drawn
+    if (!sprite.getTexture() ||
+        size.x <= 0 || size.y <= 0) {
         return;
     }
 
-    const sf::Texture& texture = *texturePtr;
-    sf::Vector2f textureSize = static_cast<sf::Vector2f>(texture.getSize());
-    const sf::IntRect& textureRect = sprite.getTextureRect();
+    // simply taken form ImGui::Image
+    ImGuiWindow* window = GetCurrentWindow();
+    if (window->SkipItems)
+        return;
+
+    ImRect itemBB(window->DC.CursorPos.x, window->DC.CursorPos.y,
+        window->DC.CursorPos.x + size.x, window->DC.CursorPos.y + size.y);
+    if (borderColor.a > 0)
+    {
+        itemBB.Max.x += 2;
+        itemBB.Max.y += 2;
+    }
+
+    ItemSize(itemBB);
+    if (!ItemAdd(itemBB, 0))
+        return;
+
+    // prepare uv coordinates
+    const sf::Vector2f textureSize = static_cast<sf::Vector2f>(sprite.getTexture()->getSize());
     ImVec2 uv0(textureRect.left / textureSize.x, textureRect.top / textureSize.y);
     ImVec2 uv1((textureRect.left + textureRect.width) / textureSize.x,
                (textureRect.top + textureRect.height) / textureSize.y);
 
-    ImTextureID textureID = convertGLTextureHandleToImTextureID(texture.getNativeHandle());
+    sf::Transform transform = additionalTransform * sprite.getTransform();
+	sf::FloatRect spriteRect = sprite.getLocalBounds();
+	const sf::FloatRect bounding = transform.transformRect(spriteRect);
+
+    // applies the transformations which are expected as a child of the parent window
+    sf::Transform childTransform;
+    childTransform.translate(itemBB.Min.x, itemBB.Min.y).
+        scale(size.x / bounding.width, size.y / bounding.height).
+        translate(-bounding.getPosition());
+    transform = childTransform * transform;
+    
+
+    if (borderColor.a > 0)
+    {
+        spriteRect.left += 1;
+        spriteRect.top += 1;
+        spriteRect.width -= 2;
+        spriteRect.height -= 2;
```

Comment:
Are the borders always 1px wide?

Reply:
Well, that's exactly what the original ``ImGui::Image`` function assumes.